pkg/parser/parser.go

```diff
@@ -0,0 +1,108 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/tjgurwara99/monkey-go/pkg/ast"
+	"github.com/tjgurwara99/monkey-go/pkg/token"
+)
+
+type Parser struct {
```

Comment:
Make this unexported - the `New` function is returning the anonymous object (which is easily inferred from the package name) with all exported things that it needs. This keeps our focus on only the things that we need.